File: study/lua/src/app/service/MailDeleteService.lua
--邮件删除服务
local BaseService = require("app.service.BaseService")
local MailDeleteService = class("MailDeleteService",BaseService)
 local UserDataHelper = require("app.utils.UserDataHelper")

function MailDeleteService:ctor()
    MailDeleteService.super.ctor(self)
    self:start()
end

function MailDeleteService:tick()
    --场景检测
    local runningSceneName = display.getRunningScene():getName() 
    if  runningSceneName ~= "main" then
        return
    end

    --判断是否开启功能
    local LogicCheckHelper = require("app.utils.LogicCheckHelper")
    local isOpen = LogicCheckHelper.funcIsOpened(FunctionConst.FUNC_MAIL)
    if not isOpen then
        return
    end

    local deleteMailIds = UserDataHelper.getExpiredMailIds()
    if #deleteMailIds > 0 then
         G_UserData:getMails():c2sDelAllMail(deleteMailIds)
    end
   
end

return MailDeleteService

File: study/ts/script/service/MailDeleteService.ts
import { BaseService } from "./BaseService";
import { G_SceneManager, G_UserData } from "../init";
import { FunctionConst } from "../const/FunctionConst";
import { LogicCheckHelper } from "../utils/LogicCheckHelper";
import { UserDataHelper } from "../utils/data/UserDataHelper";

export class MailDeleteService extends BaseService {
    
    constructor() {
        super();
        this.start();
    }
    public tick() {
        let runningSceneName = G_SceneManager.getRunningScene().getName();
        if (runningSceneName != 'main') {
            return;
        }
        let isOpen = LogicCheckHelper.funcIsOpened(FunctionConst.FUNC_MAIL)[0];
        if (!isOpen) {
            return;
        }
        let deleteMailIds = UserDataHelper.getExpiredMailIds();
        if (deleteMailIds.length > 0) {
            G_UserData.getMails().c2sDelAllMail(deleteMailIds);
        }
    }
}
